semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -86,12 +86,11 @@ trait TextDocumentOps {
           case Symbols.None =>
           case ssym =>
             samoccurrences.update(mpos, (ssym, Role.DEFINITION))
-            if (!shouldNotSaveSymbol(gsym) && !shouldNotSaveSemanticSymbol(ssym)) {
-              saveSymbolDo(sam)
-              saveSymbolFromSemantic(gsym.setInfo(sam.tpe), ssym)
+            if (!shouldNotSaveSymbol(sam) && !shouldNotSaveSemanticSymbol(ssym)) {
+              saveSymbolFromSemantic(sam, ssym)
             }
         }
-        if (gpos.isDefined && (gsym ne null)) {
+        if (gpos.isDefined && (gsym ne null) && (sam ne null)) {
```

Comment:
is it possible for `sam` to be null?

Reply:
Sure! Will do it once I get back to a computer today.